core/src/main/java/org/apache/carbondata/core/datamap/dev/DataMapFactory.java

```diff
@@ -62,6 +63,12 @@
    */
   void fireEvent(Event event);
 
+  /**
+   * Add the dataMap to cache
+   * @param dataMap
+   */
+  void addDataMapToCache(DataMap dataMap) throws IOException, MemoryException;
```

Comment:
Better add these methods in another interface CacheableDataMap and implement it to BlockletDataMapFactory directly.
```
CacheableDataMap {
  void cache(List<DataMap> dataMaps)
  List<DataMapDistributable> getAllUncachedDataMaps()
}
```